backend/checkmach.ml

```diff
@@ -219,22 +220,34 @@ end = struct
 
   let is_assume t = t.assume
 
-  let find codegen_options spec dbg =
+  let find codegen_options spec fun_name dbg =
+    let ignore_assert_all = ref false in
     let a =
       List.filter_map
         (fun (c : Cmm.codegen_option) ->
           match c with
           | Check { property; strict; assume; loc } when property = spec ->
             Some { strict; assume; loc }
-          | Check _ | Reduce_code_size | No_CSE | Use_linscan_regalloc -> None)
+          | Ignore_assert_all property when property = spec ->
+            ignore_assert_all := true;
+            None
+          | Ignore_assert_all _ | Check _ | Reduce_code_size | No_CSE
+          | Use_linscan_regalloc ->
+            None)
         codegen_options
     in
     match a with
-    | [] -> None
+    | [] ->
+      if !Clflags.zero_alloc_check_assert_all
+         && (not !ignore_assert_all)
+         && not (String.ends_with ~suffix:"__entry" fun_name)
```

Comment:
I am a little concerned that it's not completely insane for someone to name a random function this way.  I don't know anything about the entry function, but is it tricky to just figure out its exact name?  If so, I guess this is fine - I quickly checked and there are no such functions in our code base